receipt/handler.go

```diff
@@ -0,0 +1,223 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if it.Next() {
+		tx, isPending, err := eth.TransactionByHash(context.Background(), it.Event.Raw.TxHash)
+		if err != nil {
+			return nil, err
+		} else if isPending {
+			return nil, nil
+		}
+
+		from, err := types.Sender(types.LatestSignerForChainID(tx.ChainId()), tx)
+		if err != nil {
+			return nil, err
+		}
+
+		receipt, err := eth.TransactionReceipt(context.Background(), it.Event.Raw.TxHash)
+		if err != nil {
+			return nil, err
+		}
+
+		txnReceipt := &parsedTransaction{
+			BlockHash:         receipt.BlockHash,
+			BlockNumber:       hexutil.EncodeBig(receipt.BlockNumber),
+			From:              from,
+			CumulativeGasUsed: hexutil.EncodeBig(big.NewInt(0).SetUint64(receipt.CumulativeGasUsed)),
+			GasUsed:           hexutil.EncodeBig(big.NewInt(0).SetUint64(receipt.GasUsed)),
+			Logs:              receipt.Logs,
+			LogsBloom:         receipt.Bloom,
+			TransactionHash:   receipt.TxHash,
+			TransactionIndex:  hexutil.EncodeBig(big.NewInt(0).SetUint64(uint64(receipt.TransactionIndex))),
+			EffectiveGasPrice: hexutil.EncodeBig(tx.GasPrice()),
+		}
+
+		return &UserOperationReceipt{
+			Reason:        status.ProcessingStatus.String(),
+			UserOpHash:    it.Event.UserOpHash,
+			Sender:        it.Event.Sender,
+			Paymaster:     it.Event.Paymaster,
+			Nonce:         hexutil.EncodeBig(it.Event.Nonce),
+			Success:       it.Event.Success,
+			ActualGasCost: hexutil.EncodeBig(it.Event.ActualGasCost),
+			ActualGasUsed: hexutil.EncodeBig(it.Event.ActualGasUsed),
+			From:          from,
+			Receipt:       txnReceipt,
+			Logs:          []*types.Log{&it.Event.Raw},
+		}, nil
+	}
+	return nil, nil
+}
+
+// filterUserOperationEvent filters UserOperationEvents emitted by the EntryPoint contract within a specified block range.
+func filterUserOperationEvent(
+	eth *ethclient.Client,
+	userOpHash string,
+	entryPoint common.Address,
+	blkRange uint64,
+) (*entrypoint.EntrypointUserOperationEventIterator, error) {
+	// Create a new EntryPoint instance
+	ep, err := entrypoint.NewEntrypoint(entryPoint, eth)
+	if err != nil {
+		return nil, err
+	}
+
+	// Retrieve the latest block number
+	bn, err := eth.BlockNumber(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	// Calculate the block range for filtering
+	toBlk := big.NewInt(0).SetUint64(bn)
+	startBlk := big.NewInt(0)
+	if subBlkRange := big.NewInt(0).Sub(toBlk, big.NewInt(0).SetUint64(blkRange)); subBlkRange.Cmp(startBlk) > 0 {
+		startBlk = subBlkRange
+	}
+
+	// Filter UserOperationEvents within the block range
+	return ep.FilterUserOperationEvent(
+		&bind.FilterOpts{Start: startBlk.Uint64()},
+		[][32]byte{common.HexToHash(userOpHash)},
+		[]common.Address{},
+		[]common.Address{},
+	)
+}
+
+// getProcessingStatusFromDB retrieves the processing status from the database
+func getProcessingStatusFromDB(db *badger.DB, userOpHash string) (*solution.UserOpStatusMap, error) {
```

Comment:
I suggest to add it in a new package along with UpdateStatus.

Reply:
added